modules/tinymce/src/core/main/ts/api/util/ImageUploader.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Fun } from '@ephox/katamari';
+import * as Uploader from '../../file/Uploader';
+import { UploadStatus } from '../../file/UploadStatus';
+import Editor from '../Editor';
+import * as Settings from '../Settings';
+import { BlobInfo } from '../file/BlobCache';
+
+export type UploadResult = Uploader.UploadResult;
+
+interface ImageUploader {
+  upload (blobInfos: BlobInfo[], showNotification?: boolean): Promise<UploadResult[]>;
+}
+
+export const openNotification = (editor: Editor) => () => editor.notificationManager.open({
+  text: editor.translate('Image uploading...'),
+  type: 'info',
+  timeout: -1,
```

Comment:
I realise this comes from existing code and is moved, but I'm just making sure I understand; is `-1` intentional as it means the notification close button is hidden? Could we instead do:

```suggestion
  closeButton: false,
```

Reply:
Good question James. I'll leave this to the more experienced folks to answer.

Reply:
```suggestion
  timeout: -1, // TODO TINY-6679
```
as discussed in the call